app/controllers/secret_shares_controller.rb

```diff
@@ -0,0 +1,9 @@
+class SecretSharesController < ApplicationController
+
+  def create
+    moment = Moment.friendly.find(params[:moments])
+    moment.update :secret_share_identifier => SecureRandom.uuid
```

Comment:
You should take a look at `update!`; it does the same thing except it throws an exception if the update fails. `update` just silently does nothing if there is a problem.

This will upset the 👮 because it uses Ruby 1.9 syntax. To keep from trouble with the law, use `moment.update!(secret_share_identifier: SecureRandom.uuid)`

I expect this logic will only get more complex with time and so, for testabiliy, you'll want to move it into the model. In the model, you will also need to look into adding a uniqueness validator.